functions/pupil-prefs/v1.js

```diff
@@ -0,0 +1,105 @@
+const pupilColourContrastsTable = '[pupilColourContrasts]'
+const pupilTable = '[pupil]'
+const schema = '[mtc_admin]'
+
+winston.level = 'error'
+sqlService.initialise(config)
+
+const v1 = {}
+
+v1.process = async function (context, pupilPrefsMessage) {
+  const { checkCode, preferences } = pupilPrefsMessage
+  const { fontSizeCode, colourContrastCode } = preferences
+  context.log('pupil-prefs: message received', checkCode, fontSizeCode, colourContrastCode)
+
+  // Update pupil access arrangements preference based on code presence in the payload
+  try {
+    if (fontSizeCode) {
+      await v1.updatePupilAccessArrangementsPreference(checkCode, 'pupilfontSizes_id', pupilFontSizesTable, fontSizeCode, 'FTS')
+      context.log(`pupil-prefs: SUCCESS: pupil access arrangement row updated for checkCode ${checkCode} and font size code ${fontSizeCode}`)
+    }
+    if (colourContrastCode) {
+      await v1.updatePupilAccessArrangementsPreference(checkCode, 'pupilColourContrasts_id', pupilColourContrastsTable, colourContrastCode, 'CCT')
+      context.log(`pupil-prefs: SUCCESS: pupil access arrangement row updated for checkCode ${checkCode} and colour contrast code ${fontSizeCode}`)
+    }
+  } catch (error) {
+    const codeValue = fontSizeCode || colourContrastCode
+    context.log.error(`pupil-prefs: ERROR: unable to update pupil access arrangement row for ${checkCode} and code value ${codeValue}`)
+    throw error
+  }
+
+  const preparedCheckSyncQueueName = 'prepared-check-sync'
+  const message = { version: 1, checkCode: checkCode }
+  await azureStorageHelper.addMessageToQueue(preparedCheckSyncQueueName, message)
+
+  context.bindings.pupilEventsTable = []
+  const entity = {
+    PartitionKey: checkCode,
+    RowKey: uuid(),
+    eventType: 'pupil-preferences',
+    payload: JSON.stringify(pupilPrefsMessage),
+    processedAt: moment().toDate()
+  }
+  context.bindings.pupilEventsTable.push(entity)
+}
+
+/**
+ * Update relevant pupil access arrangement preference
+ * @param {String} checkCode - the unique GUID that identifies the check in the admin DB
+ * @param {String} prefField - pupil access arrangement preference field (font size or colour contrast)
+ * @param {String} prefTable - reference table to match id
+ * @param {String} prefCode - code value to be matched with reference table
+ * @param {String} accessArrangementCode - code value for relevant access arrangement
+ * @return {Promise<void>}
+ */
+v1.updatePupilAccessArrangementsPreference = async function (checkCode, prefField, prefTable, prefCode, accessArrangementCode) {
+  const sql = `UPDATE ${schema}.${pupilAccessArrangementsTable}
+               SET ${prefField} = (
+                SELECT id FROM ${schema}.${prefTable} 
+                WHERE code = @prefCode
+               )                
+               WHERE pupil_Id = (
+                  SELECT p.id FROM ${schema}.${pupilTable} p
+                  LEFT OUTER JOIN ${schema}.${checkTable} chk
```

Comment:
If we have a checkCode then shouldn't the join be guaranteed?

Reply:
@jon-shipley changed to `INNER JOIN`